VanHalenSort/VanHalenSort.ino

```diff
@@ -23,136 +24,122 @@ MQTTClient client;
     //THIS IS WHERE THE COLOR IS ACTUALLY DETECTED, EACH CHANNEL IS SMOOTHED OUT (AV. OF 5 CONSECUTIVE MEASUREMENTS) FOR MORE STABLE RESULTS
-	  client.publish("/VanHalen/color", discernColor(smoothOutFrequency(low,low), smoothOutFrequency(high,high), smoothOutFrequency(low, high)));
+    client.publish("/VanHalen/color", discernColor(new Color(smoothOutFrequency(low, low), smoothOutFrequency(high, high), smoothOutFrequency(low, high))));
     lastMillis = millis();
   }
-
 };
 
 ///////////////////////////////////////////////////////////////////////////////////
-const char* discernColor(int[] avFreqRGB){
-  //Colors tempColor = NONE;
-
-
-  int yellowRange[3][2] = {
-      {56, 60},
-      {60, 63},
-      {17, 20}
-  };
-
-
-  int redRange[3][2] = {
-      {67, 71},
-      {83, 86},
-      {21, 25}
-  };
-
-  int purpleRange[3][2] = {
-      {84, 88},
-      {84, 88},
-      {22, 25}
-  };
-
-
-  int greenRange[3][2] = {
-      {75, 80},
-      {67, 72},
-      {20, 23}
-  };
-
-   int orangeRange = {
-      {58, 62},
-      {75, 79},
-      {19, 22}
-   };
-
-  return getColorName(tempColor);
+const char *discernColor(Color c)
+{
+  for (int i = 0; i < ARRAY_SIZE(ranges); i++)
+  {
+    if (ranges[i].match(c))
+    {
+      return ranges[i].name;
+    }
+    return "NONE"
+  }
 };
 
 /////////////////TAKING AN AVERAGE FROM 5 MEASUREMENTS TO SMOOTH THE FREQ RESPONSE OUT
-int[] averageFreqRGB(){
+int[] averageFreqRGB()
+{
   int R = 0;
   int G = 0;
   int B = 0;
 
-  for(int i = 0; i < 5; i++){
+  for (int i = 0; i < 5; i++)
+  {
     //RED = low, low
-    digitalWrite(S2,LOW);
-    digitalWrite(S3,LOW);
+    digitalWrite(S2, LOW);
+    digitalWrite(S3, LOW);
     R = R + pulseIn(sensorOut, LOW);
     //GREEN = high, high
-    digitalWrite(S2,HIGH);
-    digitalWrite(S3,HIGH);
+    digitalWrite(S2, HIGH);
+    digitalWrite(S3, HIGH);
     G = G + pulseIn(sensorOut, LOW);
     //BLUE = low, high
-    digitalWrite(S2,LOW);
-    digitalWrite(S3,HIGH);
+    digitalWrite(S2, LOW);
+    digitalWrite(S3, HIGH);
     B = B + pulseIn(sensorOut, LOW);
   };
-  R = round(R/5);
-  G = round(G/5);
-  B = round(B/5);
-  return int sumRGB[3] = {R,G,B}; 
-};
-
-//////////////////////////////////////////////////////////////////////////////////////
-bool isInRange(int lowValue, int highValue, int targetValue){ 
-  boolean inRange = false;
-  if(targetValue <= highValue && targetValue >= lowValue){
-    inRange = true;
-  }
-  return inRange;
-};
-
-//////////////////////////////////GETTING THE COLOR NAMES OUT OF ENUM
-const char* getColorName(enum Colors color) {
-   switch (color) 
-   {
-      case NONE: return "none";
-      case RED: return "Red";
-      case GREEN: return "Green";
-      case PURPLE: return "Purple";
-      case YELLOW: return "Yellow";
-      case ORANGE: return "Orange";
-   }
+  R = round(R / 5);
+  G = round(G / 5);
+  B = round(B / 5);
+  return int sumRGB[3] = {R, G, B};
```

Comment:
zou een Color object moeten returnen, maar zie überhaupt niet waar het gebruikt wordt.

Reply:
nope. this is just the read out from the sensor (averaged over 5 consecuitive reads). this still has to be compared to the ranges to determine the color.